base resource install kubeEdge v1.2  , i meet two  unimaginably queer problem ; this is second problem  基于源码部署的，执行创建svc后 cloudcore进程会挂掉

1.svc.yaml文件如下：
```
apiVersion: v1
kind: Service
metadata:
  name: my-httpd-svc
  namespace: default
spec:
  clusterIP: None
  selector:
    app: my-httpd

```
2.报错日志如下：my-nginx-svc已经删除 但是后台日志一直在查看这个的状态
```
E0220 11:47:50.432364   10360 channelq.go:197] fail to get message DeletionTimestamp for message: aa7b1f18-eb16-4f50-930c-af07c00f9e77
I0220 11:47:50.432378   10360 util.go:88] The resource type is pod, 3
E0220 11:47:50.432388   10360 channelq.go:197] fail to get message DeletionTimestamp for message: 47a267ca-8d6a-4ee6-8c71-7568d2d22130
I0220 11:47:50.432401   10360 util.go:88] The resource type is service, 3
E0220 11:47:50.432412   10360 channelq.go:197] fail to get message DeletionTimestamp for message: 858ff06e-78bf-455e-870e-43ba35dae80c
I0220 11:47:50.432437   10360 util.go:88] The resource type is endpoints, 3
I0220 11:47:50.432450   10360 util.go:88] The resource type is endpoints, 3
I0220 11:47:50.432462   10360 util.go:113] The resource name is kube-controller-manager, 4
I0220 11:47:50.432150   10360 objectsync.go:116] The resourceVersion: 1164220 of service in K8s is greater than in edgenode: 1034041, send the update event
I0220 11:47:50.432501   10360 objectsync.go:116] The resourceVersion: 1164221 of endpoints in K8s is greater than in edgenode: 752253, send the update event
I0220 11:47:50.432531   10360 objectsync.go:108] endpoints: my-nginx-svc has been deleted in K8s, send the delete event to edge
E0220 11:47:50.432550   10360 objectsync.go:141] object is nil
I0220 11:47:50.432585   10360 objectsync.go:116] The resourceVersion: 1164220 of service in K8s is greater than in edgenode: 1034375, send the update event
I0220 11:47:50.432610   10360 objectsync.go:116] The resourceVersion: 859414 of endpoints in K8s is greater than in edgenode: 114804, send the update event
I0220 11:47:50.432636   10360 objectsync.go:108] service: my-nginx-svc has been deleted in K8s, send the delete event to edge
E0220 11:47:50.432655   10360 objectsync.go:141] object is nil
I0220 11:47:50.432672   10360 objectsync.go:116] The resourceVersion: 1164226 of endpoints in K8s is greater than in edgenode: 753346, send the update event
I0220 11:47:50.432696   10360 objectsync.go:108] service: my-nginx-svc has been deleted in K8s, send the delete event to edge
E0220 11:47:50.432713   10360 objectsync.go:141] object is nil
I0220 11:47:50.432726   10360 objectsync.go:116] The resourceVersion: 1164220 of service in K8s is greater than in edgenode: 753366, send the update event
I0220 11:47:50.432750   10360 objectsync.go:116] The resourceVersion: 1164218 of endpoints in K8s is greater than in edgenode: 979691, send the update event
E0220 11:47:50.432919   10360 runtime.go:78] Observed a panic: &strconv.NumError{Func:"ParseUint", Num:"", Err:(*errors.errorString)(0xc0000ae1b0)} (strconv.ParseUint: parsing "": invalid syntax)
goroutine 423 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x14509a0, 0xc0008945d0)
	/home/gowork/pkg/mod/k8s.io/apimachinery@v0.17.1/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/gowork/pkg/mod/k8s.io/apimachinery@v0.17.1/pkg/util/runtime/runtime.go:48 +0x82
panic(0x14509a0, 0xc0008945d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.CompareResourceVersion(0xc0010b4380, 0x7, 0x0, 0x0, 0x0)
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/objectsync.go:166 +0x10d
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.sendEvents(0x0, 0x0, 0xc00095e520, 0x16, 0xc000339df9, 0x7, 0xc000339e30, 0xc, 0x1622850, 0x7, ...)
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/objectsync.go:114 +0x85
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.(*SyncController).manageService(0xc000255c80, 0xc000a199c8)
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/objectsync.go:70 +0x23a
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.(*SyncController).manageObjectSync(0xc000255c80, 0xc000a42000, 0x3d, 0x40)
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/synccontroller.go:216 +0x233
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.(*SyncController).reconcile(0xc000255c80)
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/synccontroller.go:193 +0x1ea
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000ac3bc0)
	/home/gowork/pkg/mod/k8s.io/apimachinery@v0.17.1/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000ac3bc0, 0x12a05f200, 0x0, 0x1, 0xc0003f0000)
	/home/gowork/pkg/mod/k8s.io/apimachinery@v0.17.1/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000ac3bc0, 0x12a05f200, 0xc0003f0000)
	/home/gowork/pkg/mod/k8s.io/apimachinery@v0.17.1/pkg/util/wait/wait.go:88 +0x4d
created by github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.(*SyncController).Start
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/synccontroller.go:179 +0x510
panic: strconv.ParseUint: parsing "": invalid syntax [recovered]
	panic: strconv.ParseUint: parsing "": invalid syntax

goroutine 423 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/gowork/pkg/mod/k8s.io/apimachinery@v0.17.1/pkg/util/runtime/runtime.go:55 +0x105
panic(0x14509a0, 0xc0008945d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.CompareResourceVersion(0xc0010b4380, 0x7, 0x0, 0x0, 0x0)
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/objectsync.go:166 +0x10d
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.sendEvents(0x0, 0x0, 0xc00095e520, 0x16, 0xc000339df9, 0x7, 0xc000339e30, 0xc, 0x1622850, 0x7, ...)
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/objectsync.go:114 +0x85
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.(*SyncController).manageService(0xc000255c80, 0xc000a199c8)
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/objectsync.go:70 +0x23a
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.(*SyncController).manageObjectSync(0xc000255c80, 0xc000a42000, 0x3d, 0x40)
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/synccontroller.go:216 +0x233
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.(*SyncController).reconcile(0xc000255c80)
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/synccontroller.go:193 +0x1ea
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000ac3bc0)
	/home/gowork/pkg/mod/k8s.io/apimachinery@v0.17.1/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000ac3bc0, 0x12a05f200, 0x0, 0x1, 0xc0003f0000)
	/home/gowork/pkg/mod/k8s.io/apimachinery@v0.17.1/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000ac3bc0, 0x12a05f200, 0xc0003f0000)
	/home/gowork/pkg/mod/k8s.io/apimachinery@v0.17.1/pkg/util/wait/wait.go:88 +0x4d
created by github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.(*SyncController).Start
	/home/gowork/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/synccontroller.go:179 +0x510
```

--- a/cloud/pkg/synccontroller/objectsync.go
+++ b/cloud/pkg/synccontroller/objectsync.go
@@ -37,8 +37,7 @@ func (sctl *SyncController) managePod(sync *v1alpha1.ObjectSync) {
 			return
 		}
 	}
-	sendEvents(err, nodeName, sync.Namespace, sync.Spec.ObjectName, model.ResourceTypePod,
-		pod.ResourceVersion, sync.Status.ObjectResourceVersion, pod)
+	sendEvents(err, nodeName, sync, model.ResourceTypePod, pod.ResourceVersion, pod)
 }
 
 func (sctl *SyncController) manageConfigMap(sync *v1alpha1.ObjectSync) {
@@ -60,8 +59,7 @@ func (sctl *SyncController) manageConfigMap(sync *v1alpha1.ObjectSync) {
 			return
 		}
 	}
-	sendEvents(err, nodeName, sync.Namespace, sync.Spec.ObjectName, model.ResourceTypeConfigmap,
-		configmap.ResourceVersion, sync.Status.ObjectResourceVersion, configmap)
+	sendEvents(err, nodeName, sync, model.ResourceTypeConfigmap, configmap.ResourceVersion, configmap)
 }
 
 func (sctl *SyncController) manageSecret(sync *v1alpha1.ObjectSync) {
@@ -83,8 +81,7 @@ func (sctl *SyncController) manageSecret(sync *v1alpha1.ObjectSync) {
 			return
 		}
 	}
-	sendEvents(err, nodeName, sync.Namespace, sync.Spec.ObjectName, model.ResourceTypeSecret,
-		secret.ResourceVersion, sync.Status.ObjectResourceVersion, secret)
+	sendEvents(err, nodeName, sync, model.ResourceTypeSecret, secret.ResourceVersion, secret)
 }
 
 func (sctl *SyncController) manageService(sync *v1alpha1.ObjectSync) {
@@ -106,8 +103,7 @@ func (sctl *SyncController) manageService(sync *v1alpha1.ObjectSync) {
 			return
 		}
 	}
-	sendEvents(err, nodeName, sync.Namespace, sync.Spec.ObjectName, commonconst.ResourceTypeService,
-		service.ResourceVersion, sync.Status.ObjectResourceVersion, service)
+	sendEvents(err, nodeName, sync, commonconst.ResourceTypeService, service.ResourceVersion, service)
 }
 
 func (sctl *SyncController) manageEndpoint(sync *v1alpha1.ObjectSync) {
@@ -129,8 +125,7 @@ func (sctl *SyncController) manageEndpoint(sync *v1alpha1.ObjectSync) {
 			return
 		}
 	}
-	sendEvents(err, nodeName, sync.Namespace, sync.Spec.ObjectName, commonconst.ResourceTypeEndpoints,
-		endpoint.ResourceVersion, sync.Status.ObjectResourceVersion, endpoint)
+	sendEvents(err, nodeName, sync, commonconst.ResourceTypeEndpoints, endpoint.ResourceVersion, endpoint)
 }
 
 // todo: add events for devices
@@ -147,22 +142,26 @@ func (sctl *SyncController) manageDevice(sync *v1alpha1.ObjectSync) {
 	//}
 }
 
-func sendEvents(err error, nodeName, namespace, objectName, resourceType string,
-	objectResourceVersion, syncResourceVersion string,
-	obj interface{}) {
+func sendEvents(err error, nodeName string, sync *v1alpha1.ObjectSync, resourceType string,
+	objectResourceVersion string, obj interface{}) {
 
 	if err != nil && apierrors.IsNotFound(err) {
 		//trigger the delete event
-		klog.Infof("%s: %s has been deleted in K8s, send the delete event to edge", resourceType, objectName)
-		msg := buildEdgeControllerMessage(nodeName, namespace, resourceType, objectName, model.DeleteOperation, obj)
+		klog.Infof("%s: %s has been deleted in K8s, send the delete event to edge", resourceType, sync.Spec.ObjectName)
+		msg := buildEdgeControllerMessage(nodeName, sync.Namespace, resourceType, sync.Spec.ObjectName, model.DeleteOperation, obj)
 		beehiveContext.Send(commonconst.DefaultContextSendModuleName, *msg)
 		return
 	}
 
-	if CompareResourceVersion(objectResourceVersion, syncResourceVersion) > 0 {
+	if sync.Status.ObjectResourceVersion == "" {
+		klog.Errorf("The ObjectResourceVersion is empty in status of objectsync: %s", sync.Name)
+		return
+	}
+
+	if CompareResourceVersion(objectResourceVersion, sync.Status.ObjectResourceVersion) > 0 {
 		// trigger the update event
-		klog.Infof("The resourceVersion: %s of %s in K8s is greater than in edgenode: %s, send the update event", objectResourceVersion, resourceType, syncResourceVersion)
-		msg := buildEdgeControllerMessage(nodeName, namespace, resourceType, objectName, model.UpdateOperation, obj)
+		klog.Infof("The resourceVersion: %s of %s in K8s is greater than in edgenode: %s, send the update event", objectResourceVersion, resourceType, sync.Status.ObjectResourceVersion)
+		msg := buildEdgeControllerMessage(nodeName, sync.Namespace, resourceType, sync.Spec.ObjectName, model.UpdateOperation, obj)
 		beehiveContext.Send(commonconst.DefaultContextSendModuleName, *msg)
 	}
 }

